src/scoped.rs

```diff
@@ -220,32 +232,108 @@ impl<'a> fmt::Debug for Scope<'a> {
     }
 }
 
-impl<T> fmt::Debug for ScopedJoinHandle<T> {
+impl<'a, T> fmt::Debug for ScopedJoinHandle<'a, T> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "ScopedJoinHandle {{ ... }}")
     }
 }
 
 impl<'a> Scope<'a> {
+    /// Joins a spawned thread that is not joined yet.
+    ///
+    /// Returns `Some(r)` if a thread is joined and `r` is the join result, and
+    /// `None` if there are no unjoined threads.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let array = [1, 2, 3];
+    ///
+    /// crossbeam_utils::scoped::scope(|scope| {
+    ///     for i in &array {
+    ///         scope.spawn(move || {
+    ///             println!("element: {}", i);
+    ///         });
+    ///     }
+    ///
+    ///     scope.join().unwrap().unwrap();
+    ///     scope.join().unwrap().unwrap();
+    ///     scope.join().unwrap().unwrap();
+    ///     assert!(scope.join().is_none());
+    /// });
+    /// ```
+    ///
+    /// In the presence of any `ScopedJoinHandle`, you should not call `join()`
+    /// because both can take the join results of a scoped thread. In order to
+    /// forbid this case, `join()` requires a mutable reference to `Scope`. As a
+    /// result, e.g. the following code is not type-checked:
+    ///
+    /// ```ignore
+    /// let array = [1, 2, 3];
+    ///
+    /// crossbeam_utils::scoped::scope(|scope| {
+    ///     let handles = array.iter().map(|i| {
+    ///         scope.spawn(move || {
+    ///             println!("element: {}", i);
+    ///         });
+    ///     });
+    ///
+    ///     // compile error! `scope` is already borrowed above.
+    ///     scope.join();
+    /// });
+    /// ```
+    pub fn join(&mut self) -> Option<thread::Result<()>> {
+        while let Some(join) = DtorChain::pop(&mut self.joins.borrow_mut()) {
+            let ret = join.call_box();
+            if ret.is_some() {
+                return ret;
+            }
+        }
+        None
+    }
+
+    /// Joins all spawned threads that are not joined yet.
```

Comment:
Like above, it would be good to document the order of results w.r.t. the order of `spawn`s.